File: PHP/Pc.php
<?php
class Pc{
    
    private $CPU; //deklarasi Attribute
    private $RAM; //deklarasi Attribute
    private $DISK; //deklarasi Attribute
    private $total_price; //deklarasi Attribute

    #constructor 
    function __construct(Ram $RAM, Cpu $CPU, Disk $DISK, $total_price)
    {
        $this->total_price = $total_price;
        $this->RAM = $RAM;
        $this->CPU = $CPU;
        $this->DISK = $DISK;
    }
    #set get
    function setCpu(Cpu $CPU){
        $this->CPU = $CPU;
    }
    function getCpu(){
        return $this->CPU;
    }
    function setRam(Ram $RAM){
        $this->RAM = $RAM;
    }
    function getRam(){
        return $this->RAM;
    }
    function setDisk(Disk $DISK){
        $this->DISK = $DISK;
    }
    function getDisk(){
        return $this->DISK;
    }
    function setTotal_price($total_price){
        $this->total_price = $total_price;
    }
    function getTotal_price(){
        return $this->total_price;
    }
    
    #keluaran
    function printPc(){
        $this->CPU->printCpu();
        $this->RAM->printRam();
        $this->DISK->printDisk();
        echo "----- PC -----". "<br>";
        echo "Total Price: " .$this->total_price."<br>";
    }

    function __destruct()
    {
        
    }
}
?>

File: C++/pc.cpp
#include <iostream>
#include <string>
using namespace std;
// class yang akan dijadikan composite
#include "ram.cpp"
#include "disk.cpp"
#include "cpu.cpp"
class pc{
    
    private:
        ram RAM; //deklarasi Attribute
        cpu CPU; //deklarasi Attribute
        disk DISK; //deklarasi Attribute
        int total_price; //deklarasi Attribute
    public:
        // Constructor
        pc(){

        }

        pc(ram a, cpu b, disk c, int total){
            this->RAM = a;
            this->CPU = b;
            this->DISK = c;
            this->total_price = total;
        }

        // Setter Getter
        void setRam(ram a){ 
            this->RAM = a; 
        }
        ram getRam(){ 
            return this->RAM; 
        }
        void setCpu(cpu b){ 
            this->CPU = b;
        }
        cpu getCpu(){ 
            return this->CPU; 
        }
        void setDisk(disk c){ 
            this->DISK = c;
        }
        disk getDisk(){ 
            return this->DISK; 
            }
        void setTotal(int total){ 
            this->total_price = total;
        }
        int getTotal(){ 
            return this->total_price;
        }
        
        //keluaran
        void printPC(){
            this->RAM.printRam();
            this->CPU.printCpu();
            this->DISK.printDisk();
            cout << "---- PC ----" << endl;
            cout << "Total Price: " << this->total_price << endl;
        }
        ~pc(){}
};